File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

// FIXME: This example is incomplete

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your LED Strip Bricklet 2.0

	// Use frame started callback to move the active LED every frame
	static void FrameStartedCB(BrickletLEDStripV2 sender, int length)
	{
		Console.WriteLine("Length: " + length);
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletLEDStripV2 ls = new BrickletLEDStripV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Set frame duration to 50ms (20 frames per second)
		ls.SetFrameDuration(50);

		// Register frame started callback to function FrameStartedCB
		ls.FrameStartedCallback += FrameStartedCB;

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

# FIXME: This example is incomplete

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_led_strip_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your LED Strip Bricklet 2.0

ipcon = IPConnection.new # Create IP connection
ls = BrickletLEDStripV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Set frame duration to 50ms (20 frames per second)
ls.set_frame_duration 50

# Use frame started callback to move the active LED every frame
ls.register_callback(BrickletLEDStripV2::CALLBACK_FRAME_STARTED) do |length|
  puts "Length: #{length}"
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
